app/Repositories/BatchRepository.php

```diff
@@ -57,4 +101,33 @@ public function update(array $data, $id)
 
         return response()->json(['message' => config('api.update')]);
     }
+
+    public function destroy($id)
+    {
+        DB::beginTransaction();
+        try {
+            $batch = Batch::findOrFail($id);
+            $users = $batch->users;
+            foreach ($users as $user) {
+                $user->dayMonths()->detach();
+                User::destroy($user->id);
```

Comment:
ko nên gọi model trong vòng lặp nhé
=> https://laracasts.com/discuss/channels/eloquent/how-to-delete-multiple-records-using-laravel-eloquent